test/integ/test_put_get_compress_enc.py

```diff
@@ -0,0 +1,228 @@
+
+import pytest
+
+from ..integ_helpers import put
+
+pytestmark = pytest.mark.skipolddriver  # old test driver tests won't run this module
+
+try:
+    from snowflake.connector.s3_storage_client import SnowflakeS3RestClient
+
+    orig_send_req = SnowflakeS3RestClient._send_request_with_authentication_and_retry
+    hack_s3_rest_client = True
+except ImportError:
+    hack_s3_rest_client = False
+    SnowflakeS3RestClient = None
+    orig_send_req = None
+
+
+THIS_DIR = path.dirname(path.realpath(__file__))
+
+logger = getLogger(__name__)
+
+
+def _prepare_tmp_file(to_dir: str) -> str:
+    tmp_dir = to_dir / "data"
+    tmp_dir.mkdir()
+    file_name = "data.txt"
+    test_path = tmp_dir / file_name
+    with test_path.open("w") as f:
+        f.write("test1,test2\n")
+        f.write("test3,test4")
+    return test_path, file_name
+
+
+def mock_send_request(
+    self,
+    url,
+    verb,
+    retry_id,
+    query_parts=None,
+    x_amz_headers=None,
+    headers=None,
+    payload=None,
+    unsigned_payload=False,
+    ignore_content_encoding=False,
+):
+    # when called under _initiate_multipart_upload and _upload_chunk, add content-encoding to header
+    if verb is not None and verb in ("POST", "PUT") and headers is not None:
+        headers["Content-Encoding"] = "gzip"
+    return orig_send_req(
+        self,
+        url,
+        verb,
+        retry_id,
+        query_parts,
+        x_amz_headers,
+        headers,
+        payload,
+        unsigned_payload,
+        ignore_content_encoding,
+    )
+
+
+@pytest.mark.skipif(hack_s3_rest_client, reason="s3_rest_client could not be imported.")
+@pytest.mark.parametrize("auto_compress", [True, False])
+def test_auto_compress_switch(
+    tmp_path: pathlib.Path,
+    conn_cnx,
+    auto_compress,
+):
+    """Tests PUT command with auto_compress=False|True."""
+    _test_name = "test_auto_compress_switch"
+    test_data, file_name = _prepare_tmp_file(tmp_path)
+
+    with conn_cnx() as cnx:
+        cnx.cursor().execute(f"RM @~/{_test_name}")
+        try:
+            file_stream = test_data.open("rb")
+            with cnx.cursor() as cur:
+                put(
+                    cur,
+                    str(test_data),
+                    f"~/{_test_name}",
+                    False,
+                    sql_options=f"auto_compress={auto_compress}",
+                    file_stream=file_stream,
+                )
+
+            ret = cnx.cursor().execute(f"LS @~/{_test_name}").fetchone()
+            uploaded_gz_name = f"{file_name}.gz"
+            if auto_compress:
+                assert uploaded_gz_name in ret[0]
+            else:
+                assert uploaded_gz_name not in ret[0]
+
+            # get this file, if the client handle compression meta correctly
+            get_dir = tmp_path / "get_dir"
+            get_dir.mkdir()
+            cnx.cursor().execute(
+                f"GET @~/{_test_name}/{file_name} file://{get_dir}"
+            ).fetchone()
```

Comment:
This doesn't do anything in this test
```suggestion
            )
```